Source/WebInspectorUI/UserInterface/Base/Main.js

```diff
@@ -237,6 +237,12 @@ WI.contentLoaded = function()
     WI.settings.resourceCachingDisabled.addEventListener(WI.Setting.Event.Changed, WI._resourceCachingDisabledSettingChanged, WI);
     WI.settings.experimentalAllowInspectingInspector.addEventListener(WI.Setting.Event.Changed, WI._allowInspectingInspectorSettingChanged, WI);
 
+    if (WI.engineeringSettingsAllowed()) {
+        WI.settings.engineeringShowSiteIsolationDebugOverlay.addEventListener(WI.Setting.Event.Changed, function() {
+            WI.mainTarget.PageAgent.overrideSetting(InspectorBackend.Enum.Page.Setting.SiteIsolationDebugOverlay, WI.settings.engineeringShowSiteIsolationDebugOverlay.value);
+        });
```

Comment:
`WI.Object.prototype.addEventListener` will `console.assert()` if you dont provide a 3rd argument `thisObject` (which in this case id recommend being `WI.settings.engineeringShowSiteIsolationDebugOverlay` since that's the thing being listened for and to match what's done elsewhere)